File: native-micronaut-api/src/main/java/com/xavierbouclet/nativeapi/Actor.java
package com.xavierbouclet.nativeapi;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.Set;
import java.util.UUID;

@Entity
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Actor {

    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private UUID id = null;

    private String fullName = "";

    @JsonIgnoreProperties("actors")
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "actor_movie_mapping",
            joinColumns = @JoinColumn(name = "actor_id", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "movie_id", referencedColumnName = "id"))
    private Set<Movie> movies;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }

    public Set<Movie> getMovies() {
        return movies;
    }

    public void setMovies(Set<Movie> movies) {
        this.movies = movies;
    }
}

File: native-micronaut-api/src/main/kotlin/com/xavierbouclet/nativeapi/Actor.kt
package com.xavierbouclet.nativeapi

import com.fasterxml.jackson.annotation.JsonIdentityInfo
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.ObjectIdGenerators
import io.micronaut.data.annotation.Id
import org.hibernate.annotations.GenericGenerator
import java.util.*
import javax.persistence.*

@Entity
@JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator::class, property="id")
class Actor {

    @javax.persistence.Id
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    var id: UUID? = null

    var fullName: String = ""

    @JsonIgnoreProperties("actors")
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "actor_movie_mapping",
    joinColumns = [JoinColumn(name = "actor_id", referencedColumnName = "id")],
    inverseJoinColumns = [JoinColumn(name = "movie_id", referencedColumnName = "id")])
    var movies: MutableSet<Movie>? = mutableSetOf()
}
